pkg/instancetype/instancetype.go

```diff
@@ -886,13 +891,20 @@ func applyCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.Virtua
 	vmiSpec.Domain.CPU.Sockets = 1
 	vmiSpec.Domain.CPU.Threads = 1
 
+	// PreferSpreadDivider is a constant used for dividing when PreferSpread is selected.
+	const PreferSpreadDivider = 2
+
 	switch getPreferredTopology(preferenceSpec) {
 	case instancetypev1beta1.PreferCores:
 		vmiSpec.Domain.CPU.Cores = instancetypeSpec.CPU.Guest
 	case instancetypev1beta1.PreferSockets:
 		vmiSpec.Domain.CPU.Sockets = instancetypeSpec.CPU.Guest
 	case instancetypev1beta1.PreferThreads:
 		vmiSpec.Domain.CPU.Threads = instancetypeSpec.CPU.Guest
+	case instancetypev1beta1.PreferSpread:
+		vmiSpec.Domain.CPU.Sockets = instancetypeSpec.CPU.Guest / PreferSpreadDivider
+		// Apply the same ratio to cores but also add any left over vCPUs
+		vmiSpec.Domain.CPU.Cores = vmiSpec.Domain.CPU.Sockets + (instancetypeSpec.CPU.Guest % PreferSpreadDivider)
```

Comment:
This will end up with more CPUs than initially requested? E.g. 5 vCPUs with spread topology will lead to 2 sockets with 3 cores each?

Reply:
Yes, it will be set to 3 cores and 2 sockets. 

What do you think about that @lyarwood ?

Reply:
@0xFelix is correct, 5 would up providing 6 (2 * 3) vCPUs technically.

We don't have a way of expressing 2 sockets with 2 cores and 1 with 1 so I'm not sure how we could spread this other than 5 sockets with 1 core?